rest/model/conversions.go

```diff
@@ -86,44 +95,50 @@ func objectTypeFnName(in types.Type, outIsPtr bool) (convertFuncs, error) {
 	return funcs, nil
 }
 
-func simplePtrs(typeName string, outIsPtr bool, generatedConversions map[string]string) (convertFuncs, error) {
+func simplePtrs(inType, outType string, inIsPtr, outIsPtr bool, generatedConversions map[typeInfo]string) (convertFuncs, error) {
 	funcs := convertFuncs{}
-	if err := generateConversionCode(generatedConversions, typeName); err != nil {
+	if err := generateConversionCode(generatedConversions, inType, outType); err != nil {
 		return funcs, err
 	}
-	typeName = cleanName(typeName)
-	if outIsPtr {
-		funcs.converter = fmt.Sprintf("%s%sPtr", typeName, typeName)
-		funcs.inverter = fmt.Sprintf("%sPtr%s", typeName, typeName)
-		return funcs, nil
+	if inType != outType {
+		if err := generateConversionCode(generatedConversions, outType, inType); err != nil {
+			return funcs, err
+		}
 	}
-	funcs.converter = fmt.Sprintf("%s%s", typeName, typeName)
-	funcs.inverter = funcs.converter
+	inType = cleanName(inType, inIsPtr)
+	outType = cleanName(outType, outIsPtr)
+	funcs.converter = fmt.Sprintf("%s%s", inType, outType)
+	funcs.inverter = fmt.Sprintf("%s%s", outType, inType)
 	return funcs, nil
 }
 
-func cleanName(name string) string {
+func cleanName(name string, addPtr bool) string {
```

Comment:
This is the name of the variable, not the type of the variable, right? I'm confused why this is being used for inType / outType.

Reply:
this is part of the name for a function, and the types of the variables are in the name. Added some comments explaining things